### ISSUE TITLE
Make apimcli output easy to script

### ISSUE THREAD

**Issue by chintana** [wso2/product-apim-tooling#49]
**Description:**
Current output format used in apimcli is not easy to script. For example, one use case for this is to export a list of APIs. For that we would have to iterate through the list of APIs and call export for each and every API. The list out put at the moment looks like below,

```
Environment: local
No. of APIs: 6
+-----------+---------+---------------+-----------+----------+--------------------------------------+
|   NAME    | VERSION |    CONTEXT    |  STATUS   | PROVIDER |                  ID                  |
+-----------+---------+---------------+-----------+----------+--------------------------------------+
| Claims    | v1      | /claims       | PUBLISHED | admin    | 2db0e25e-c162-4905-86e3-88e9e3795d1b |
| Claims    | v2      | /claims       | PUBLISHED | admin    | 6d32a1df-59eb-4327-be7b-1ba7d5cbcb2e |
| Claims    | v3      | /claims       | PUBLISHED | admin    | 86035f02-2d09-4602-980a-4c2774b8ed3f |
| Providers | v1      | /providers    | PUBLISHED | admin    | ca336dec-34be-4715-83b3-4cf65650c30c |
| Providers | v2      | /providers    | PUBLISHED | admin    | 2ee35727-3f96-4fc0-b7e2-52d82bdc02fa |
| Providers | v3      | /providers    | PUBLISHED | admin    | 6c9d865b-16fe-4393-8f3c-878feff83f54 |
+-----------+---------+---------------+-----------+----------+--------------------------------------+
```

First, the script should ignore the first 4-5 lines, then ignore special table delimiter lines with +---+---+.  This is a bit cumbersome to do in a bash script. After that we need to take each line, and get name, version, context and pass it to export command.

Is it possible to make the output easy to script? Similar to output given by docker for example. That's very easy to chain multiple commands and script.
**Affected Product Version:**
APIM CLI 1.2.0, APIM 2.6.0

**Comment by menuka94** [comment]
I did a preliminary fix in #50 and now the output is as follows. How does this look?
```
Environment: dev
No. of APIs: 3
  NAME      VERSION  CONTEXT  STATUS      PROVIDER  ID                                    

  TestAPI   1.0.0    /test    PROTOTYPED  admin     b3f6de4e-f14f-440e-aac7-280b85c7c2ce  
  TestAPI2  1.2.0    /test2   PROTOTYPED  admin     e3c07cae-5623-4df8-a50e-4d55f02c62f6  
  TestAPI3  2.0.0    /test3   CREATED     admin     73b77b0a-1fcb-4105-9cbd-21b9d717af40  

```

**Comment by chintana** [comment]
@menuka94 is it possible to remove the first 2 lines? Then it's relatively easier .. also is there a plan to format the output? Similar to $ docker ps --format "{{.ID}}: {{.Command}}"

For systems with a lot of APIs, may be we can give the option of displaying labels as well? Then it's easy to filter by a key=value and import/export a subset of APIs

**Comment by kasvith** [comment]
Yes, that can be done :) We can give a `--format` flag to get an arbitrary output from the command.

**Comment by kasvith** [comment]
@chintana the following #66 fix the issue partially. It currently does not support custom table formatted outputs like in docker

It now supports `json`, `jsonPretty` formats and using go templates users can print it using any custom format.

**Comment by kasvith** [comment]
Closed PR for enhancing the feature, we may be can get docker like output with custom table support too :) will reopen soon, Then it will close this issue

**Comment by kasvith** [comment]
@chintana I updated the PR and now it fully supports docker like table formatting too. Please take a look.

**Comment by kasvith** [comment]
the issue can be closed as #66 was merged

**Comment by harsha89** [comment]
Done thanks for your contribution
